### PR TITLE
Quick Menu: Keyboard functionality

### LINKED ISSUE #7509: Quick Menu: Keyboard functionality
Quick menu is a context menu that's always visible. 

This ticket is to make sure the keyboard functionality is 100% now that we can see it in the editor with its many focus traps. 

Right now, focus gets trapped in the menu, we should be able to continue tabbing through the page after the last menu item and also backwards through focus with tab + shift. 

(not really a bug since this feature is in flight, just a more specific ticket to isolate changes since it wasn't an issue in storybook)

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
More work around integrating quick menu into the editor. This ticket focuses on better keyboard navigation.

## Summary
Prior to this, wasn't possible to keyboard navigate in and out of the quick menu although it have full keyboard navigability once the focus was within it. This ticket fixes that and integrates its keyboard navigation flow into the rest of the canvas navigation layer.

## Relevant Technical Choices
Moved quick menu into the Nav Layer to work with the platform as far as keyboard navigation order. Doesn't change appearance at all.

## To-do
NA

## User-facing changes
NA

## Testing Instructions
Click on the title input so it receives focus, tab to the submit button, press tab again and you should be within the quick menu, tabbing to the last element and pressing tab again should take you to the first focusable element in the carousel region of the nav layer.

### QA
Enable Quick menu in experiments. follow testing instructions above and see that it works.

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7509

### CODE REVIEW

File: assets/src/design-system/components/contextMenu/menu.js [hunk 1]
```diff
@@ -207,7 +214,7 @@ const Menu = ({ items, isIconMenu, isOpen, onDismiss, ...props }) => {
       // If we didn't find a focusable element or get to the start/end
       // of the list then **tabbing should close the menu**
       if (key === KEYS.TAB) {
-        onDismiss?.();
+        onDismiss?.({ isAscending });
```
Comment (samwhale) [inline assets/src/design-system/components/contextMenu/menu.js]: Do we need this? I don't think this is used anywhere

File: assets/src/edit-story/constants/index.js [hunk 2]
```diff
@@ -95,4 +95,15 @@ export const HIDDEN_PADDING = {
...
+export const FOCUSABLE_SELECTORS = [
+  'button',
+  '[href]',
+  'input',
+  'select',
+  'textarea',
+  '[tabindex]:not([tabindex="-1"])',
+];
+
+export const ALL_FOCUSABLE_SELECTOR = FOCUSABLE_SELECTORS.join(', ');
```
Comment (samwhale) [inline assets/src/edit-story/constants/index.js]: I don't think we use this either, right?

File: assets/src/edit-story/components/canvas/navLayer.js [hunk 3]
```diff
@@ -17,26 +17,53 @@
...
+  const quickActions = useQuickActions();
+
   return (
     <Layer
       pointerEvents="none"
       zIndex={Z_INDEX.NAV}
       onMouseDown={(evt) => evt.stopPropagation()}
     >
-      <HeadArea pointerEvents="initial">
+      <HeadArea pointerEvents="initial" ref={headAreaRef}>
```
Comment (samwhale) [inline assets/src/edit-story/components/canvas/navLayer.js]: 🤔 what are we using the refs for?

**Review by samwhale — APPROVED** [review]
Totally works - smart move moving this up to the nav layer

The context menu can be very annoying due to the custom focus stuff. This may have already been here but I noticed that tabbing out of the context menu (ascending + descending) in the dashboard ends up focusing the body instead of going back to the menu button
tab-to-body

I think we should make a ticket to update that in the future. It's not a big deal

File: assets/src/edit-story/components/canvas/navLayer.js (hunk 3 above)
Comment (maxyinger) [reply assets/src/edit-story/components/canvas/navLayer.js]: Oh good call outs. Let me clean up a bit. That was all for a prior approach where the tab functionality was all controlled

Comment (maxyinger) [reply assets/src/edit-story/components/canvas/navLayer.js]: eventually figured If I'm recreating native tab functionality, I should just use native tab functionality lol

**Comment by maxyinger** [comment]
> The context menu can be very annoying due to the custom focus stuff. This may have already been here but I noticed that tabbing out of the context menu (ascending + descending) in the dashboard ends up focusing the body instead of going back to the menu button

Yea I think this would have been there before since I didn't alter any backwards functionality (that I know of). Happy to make a ticket for this although I'm not sure what the ticket would be?

_note:_ confirmed this is the existing behavior on main and not introduced by this PR

**Comment by samwhale** [comment]
> > The context menu can be very annoying due to the custom focus stuff. This may have already been here but I noticed that tabbing out of the context menu (ascending + descending) in the dashboard ends up focusing the body instead of going back to the menu button
> 
> Yea I think this would have been there before since I didn't alter any backwards functionality (that I know of). Happy to make a ticket for this although I'm not sure what the ticket would be?
> 
> _note:_ confirmed this is the existing behavior on main and not introduced by this PR

Awesome, yeah lets tackle that in a different ticket <3

**Review by BrittanyIRL — APPROVED** [review]
Nice!

**Comment by BrittanyIRL** [comment]
@googlebot I consent.

**Pull request merged**